dspace-api/src/main/java/org/dspace/app/itemimport/ItemImportServiceImpl.java

```diff
@@ -1518,28 +1676,62 @@ protected void processOptions(Context c, Item myItem, List<String> options)
+                            + " not found, permissions set to default");
                     } else if (actionID == -1) {
                         System.out
                             .println("\tinvalid permissions flag, permissions set to default");
                     } else {
                         System.out.println("\tSetting special permissions for "
-                                               + bitstreamName);
+                            + bitstreamName);
                         setPermission(c, myGroup, actionID, bs);
                     }
                 }
 
                 if (descriptionExists) {
                     System.out.println("\tSetting description for "
-                                           + bitstreamName);
+                        + bitstreamName);
                     bs.setDescription(c, thisDescription);
                     bitstreamService.update(c, bs);
                 }
+
+                if (labelExists) {
+                    MetadataField metadataField = metadataFieldService
+                        .findByElement(c, "iiif", "label", null);
+                    System.out.println("\tSetting label to " + thisLabel + " in element "
+                        + metadataField.getElement() + " on " + bitstreamName);
+                    bitstreamService.addMetadata(c, bs, metadataField, null, thisLabel);
+                    bitstreamService.update(c, bs);
+                }
+
+                if (heightExists) {
+                    MetadataField metadataField = metadataFieldService
+                        .findByElement(c, "iiif", "image", "height");
+                    System.out.println("\tSetting height to " + thisHeight + " in element "
+                        + metadataField.getElement() + " on " + bitstreamName);
+                    bitstreamService.addMetadata(c, bs, metadataField, null, thisHeight);
+                    bitstreamService.update(c, bs);
+                }
+                if (widthExists) {
+                    MetadataField metadataField = metadataFieldService
+                        .findByElement(c, "iiif", "image", "width");
+                    System.out.println("\tSetting width to " + thisWidth + " in element "
+                        + metadataField.getElement() + " on " + bitstreamName);
+                    bitstreamService.addMetadata(c, bs, metadataField, null, thisWidth);
+                    bitstreamService.update(c, bs);
+                }
+                if (tocExists) {
+                    MetadataField metadataField = metadataFieldService
+                        .findByElement(c, "iiif", "toc", null);
+                    System.out.println("\tSetting toc to " + thisToc + " in element "
+                        + metadataField.getElement() + " on " + bitstreamName);
+                    bitstreamService.addMetadata(c, bs, metadataField, null, thisToc);
+                    bitstreamService.update(c, bs);
```

Comment:
perform the update just one time at the end of the method if at least one bitstream metadata has been added